lib/grammars.coffee

```diff
@@ -137,7 +137,11 @@ module.exports =
   Go:
     "File Based":
       command: "go"
-      args: (context) -> ['run', context.filepath]
+        args: (context) ->
+          if context.filepath.match(/_test.go/)
+            ['test', '']
+          else ['run', context.filepath]
+      workingDirectory: atom.workspace.getActivePaneItem()?.buffer?.file?.getParent?().getPath?()
```

Comment:
I try to use as working directory the one where the file is placed.

Francisco Javier Zorzano Mier
Physical Internet Lab
Telefónica/CCDO Org/Internet of Things/Technology&Platforms/
+34 630538546
@zorzano, zorzano@tid.es<mailto:zorzano@tid.es>, skype: zorzano

De: Kyle Kelley [mailto:notifications@github.com]
Enviado el: viernes, 15 de abril de 2016 16:47
Para: rgbkrk/atom-script
CC: FRANCISCO JAVIER ZORZANO MIER
Asunto: Re: [rgbkrk/atom-script] Adapt Script for tests in go (#886)


In lib/grammars.coffee<https://github.com/rgbkrk/atom-script/pull/886#discussion_r59885615>:

> @@ -137,7 +137,11 @@ module.exports =

>    Go:

>      "File Based":

>        command: "go"

> -      args: (context) -> ['run', context.filepath]

> +        args: (context) ->

> +          if context.filepath.match(/_test.go/)

> +            ['test', '']

> +          else ['run', context.filepath]

> +      workingDirectory: atom.workspace.getActivePaneItem()?.buffer?.file?.getParent?().getPath?()

I'm a bit unsure what you're trying to do here. Is this reaching the root of the project for the working Directory?

—
You are receiving this because you authored the thread.
Reply to this email directly or view it on GitHub<https://github.com/rgbkrk/atom-script/pull/886/files/5f233b612e82dfdb48c8343603f2cb46b7a99f03#r59885615>

________________________________

Este mensaje y sus adjuntos se dirigen exclusivamente a su destinatario, puede contener información privilegiada o confidencial y es para uso exclusivo de la persona o entidad de destino. Si no es usted. el destinatario indicado, queda notificado de que la lectura, utilización, divulgación y/o copia sin autorización puede estar prohibida en virtud de la legislación vigente. Si ha recibido este mensaje por error, le rogamos que nos lo comunique inmediatamente por esta misma vía y proceda a su destrucción.

The information contained in this transmission is privileged and confidential information intended only for the use of the individual or entity named above. If the reader of this message is not the intended recipient, you are hereby notified that any dissemination, distribution or copying of this communication is strictly prohibited. If you have received this transmission in error, do not read it. Please immediately reply to the sender that you have received this communication in error and then delete it.

Esta mensagem e seus anexos se dirigem exclusivamente ao seu destinatário, pode conter informação privilegiada ou confidencial e é para uso exclusivo da pessoa ou entidade de destino. Se não é vossa senhoria o destinatário indicado, fica notificado de que a leitura, utilização, divulgação e/ou cópia sem autorização pode estar proibida em virtude da legislação vigente. Se recebeu esta mensagem por erro, rogamos-lhe que nos o comunique imediatamente por esta mesma via e proceda a sua destruição